java/src/jmri/jmrix/loconet/LocoNetThrottle.java

```diff
@@ -549,11 +551,22 @@ public void notifyChangedSlot(LocoNetSlot pSlot) {
         }
 
         // Functions
+        updateFunctions();
+
+        log.debug("notifyChangedSlot ends");
+    }
+
+    /**
+     * update the F0-F29 functions.
+     * Invoked by notifyChangedSlot(), this nominally updates from the slot.
+     */
+    protected void updateFunctions() {
         for (int i = 0; i < 29; i++) {
+            log.debug("updateFunction({}, {})", i, slot.isFunction(i));
+            if (i==20 && log.isDebugEnabled()) log.trace("Tracing back F20", new Exception("traceback"));
```

Comment:
Is this line intended to be included in master?

Reply:
Yes.  The IB uses some custom messages that sometimes follow really odd paths though the LocoNet code.  This PR resolves some of those, but I'm pretty sure that the F12 and F20 messages are still iffy.  If somebody encountered an error with real hardware, I'd like to be able to turn this on with TRACE.

Reply:
But it probably should be isTraceEnabled(), so will fix that.